src/SourceBuild/content/test/Microsoft.DotNet.Tests/PackageSourceMappingsTests.cs

```diff
@@ -0,0 +1,242 @@
+            string expectedNugetConfig = Path.Combine(psmAssetsDir, "expected", nugetConfigFilename);
+
+            string modifiedNugetConfig = Path.Combine(PackageSourceMappingsSetup.PackageSourceMappingsRoot, nugetConfigFilename);
+            Directory.CreateDirectory(Path.GetDirectoryName(modifiedNugetConfig)!);
+            File.Copy(originalNugetConfig, modifiedNugetConfig, true);
+            UpdateNugetConfigTokens(modifiedNugetConfig);
+
+            var task = new UpdateNuGetConfigPackageSourcesMappings()
+            {
+                SbrpCacheSourceName = "source-build-reference-package-cache",
+                SbrpRepoSrcPath = TestSetup.SourceBuildReferencePackagesRepoDir,
+                SourceBuiltSourceNamePrefix = "source-built-",
+                NuGetConfigFile = modifiedNugetConfig,
+                BuildWithOnlineFeeds = useOnlineFeeds,
+                SourceBuildSources = sources,
+                CustomSources = customSources
+            };
+
+            if (sourceBuild)
+            {
+                task.ReferencePackagesSourceName = "reference-packages";
+                task.PreviouslySourceBuiltSourceName = "previously-source-built";
+                task.PrebuiltSourceName = "prebuilt";
+            }
+
+            task.Execute();
+
+            TokenizeLocalNugetConfigFeeds(modifiedNugetConfig);
+
+            string expectedNugetConfigContents = File.ReadAllText(expectedNugetConfig);
+            string modifiedNugetConfigContents = File.ReadAllText(modifiedNugetConfig);
+            Assert.Equal(expectedNugetConfigContents, modifiedNugetConfigContents);
+        }
+
+        private static void UpdateNugetConfigTokens(string nugetConfigFile)
+        {
+            string fileContents = File.ReadAllText(nugetConfigFile);
+            foreach (KeyValuePair<string, string> kvp in TestSetup.LocalTokenSourceMappings)
+            {
+                fileContents = fileContents.Replace(kvp.Key, kvp.Value);
+            }
+            File.WriteAllText(nugetConfigFile, fileContents);
+        }
+
+        private static void TokenizeLocalNugetConfigFeeds(string nugetConfigFile)
+        {
+            string fileContents = File.ReadAllText(nugetConfigFile);
+            foreach (KeyValuePair<string, string> kvp in TestSetup.LocalTokenSourceMappings)
+            {
+                fileContents = fileContents.Replace(kvp.Value, kvp.Key);
+            }
+            File.WriteAllText(nugetConfigFile, fileContents);
+        }
+
+        internal class PackageSourceMappingsSetup
+        {
+            private static PackageSourceMappingsSetup? instance;
+            private static readonly object myLock = new();
+            private Dictionary<string, string>? localTokenSourceMappings;
+
+            public static readonly string PackageSourceMappingsRoot = Path.Combine(Directory.GetCurrentDirectory(), nameof(PackageSourceMappingsTests));
```

Comment:
How does this work it you run the tests multiple times without "cleaning"?  Should this be a unique name?

Reply:
Yeah, I have a fix for this, will always use a new temp folder.

Reply:
Fixed with https://github.com/dotnet/sdk/commit/6337002902416768d8776d865625b83fd6d75d79